src/control/lib/hardware/fabric.go

```diff
@@ -136,7 +314,7 @@ func (s fabricInterfaceMap) update(name string, fi *FabricInterface) {
 		// always possible to add to providers or net interfaces
 		if fi.Providers != nil {
 			if cur.Providers == nil {
-				cur.Providers = common.NewStringSet()
+				cur.Providers = newTestFabricProviderSet()
```

Comment:
as this is just creating an empty set, use NewFabricProviderSet() and remove newTestFabricProviderSet() function completely

Reply:
Good catch, there was a bit of rename shuffling here. I'll fix this.